### ISSUE TITLE
[bug]: `no-useless-fragment` False positive when passing a ref to a Fragment

### ISSUE THREAD

**Issue by sadan4** [Rel1cx/eslint-react#1567]
### Describe the bug

There is a false positive while passing a ref prop to a Fragment to use a FragmentInstance (https://react.dev/reference/react/Fragment#fragmentinstance), as only the key property is checked and not the ref property, see: https://github.com/Rel1cx/eslint-react/blob/8d92c0c5384fe5e4ad64e23dead2859561f35c2e/packages/plugins/eslint-plugin-react-x/src/rules/no-useless-fragment/no-useless-fragment.ts#L78-L81

In addition, the auto-fix for this is incorrect. By removing the Fragment, the ref is no longer set, producing *incorrect code without any errors*.

### Reproduction

```tsx
import { Fragment, useRef } from "react";

export function Foo({ children }) {
    const ref = useRef();
    return (
        <div>
            {/* ... */}
            <Fragment ref={ref}>
                {children}
            </Fragment>
            {/* ... */}
        </div>
    );
}
```

### Expected behavior

no lint raised on the above code;

### Platform and versions

```Text
`"@eslint-react/eslint-plugin": "3.0.0-beta.57"`
```

### Stack trace

```Text

```
